### DISCUSSION TITLE
Import radio show titles from TuneIn radio stations which support this

### DISCUSSION THREAD

**Discussion by nagyrobi** [music-assistant/support#455] [Feature requests and ideas]
Some radios on TuneIn have their show run schedules uploaded to TuneIn, and on its web interface it is shown:

![image](https://user-images.githubusercontent.com/1550668/168161161-dd3a8794-4f04-4a3a-a499-f54161650ed6.png)

Would be nice to import that too while playing in the event bus so we can display it if there's no Artist / Title information available otherwise from ICY or other sources.

Screenshot above is for Rádió 1 Budapest `"provider": "tunein", "item_id": "s15879"`.

**Comment by nagyrobi** [comment]
Might wanna only import the current and the next show, and whenever the next is reached in time, refresh the data, so we get the next-next...

**Comment by erkr** [comment]
relates to issue https://github.com/music-assistant/hass-music-assistant/issues/126

**Comment by stefan-sherwood** [comment]
The TuneIn show info appears in HomeAssistant when casting to Google Home devices but only when initiating the stream by voice, e.g.:

"hey Google, play WWOZ"
(response) "Playing WWOZ on TuneIn"

but not when playing through Music Assistant.

**Comment by OzGav** [comment]
Support for radio station metadata when streamed via HLS will be added to 2.8.0beta 7. Support for older ICYMetadata streams is awaiting approval.
